apps/api/pages/api/teams/[teamId]/_auth-middleware.ts

```diff
@@ -23,6 +23,16 @@ export async function checkPermissions(
 ) {
   const { userId, prisma, isAdmin } = req;
   const { teamId } = schemaQueryTeamId.parse(req.query);
+  return canUserAccessTeamWithRole(prisma, userId, isAdmin, teamId, role);
+}
+
+export async function canUserAccessTeamWithRole(
```

Comment:
My hope is that this can be used ad a default for checking team access rather than just defaulting to checking if the user is an admin 🙏